src/vivarium_public_health/plugins/parser.py

```diff
@@ -334,3 +333,407 @@ def get_data_getter(
+                error_messages += self._validate_state(cause_name, state_name, state_config)
+
+        transitions_config = cause_config.get("transitions", {})
+        can_parse_transitions = bool(transitions_config)
+        if not isinstance(transitions_config, dict):
+            can_parse_transitions = False
+            error_messages.append(
+                f"Transitions configuration for cause '{cause_name}' must be "
+                "a dictionary if it is present."
+            )
+
+        if can_parse_transitions:
+            for _, transition_config in cause_config["transitions"].items():
+                error_messages += self._validate_transition(
+                    cause_name, transition_config, states_config
+                )
+
+        return error_messages
+
+    def _validate_state(
+        self, cause_name: str, state_name: str, state_config: Dict[str, Any]
+    ) -> List[str]:
+        """
+        Validates a state configuration and returns a list of error messages.
+
+        Parameters
+        ----------
+        cause_name
+            The name of the cause to which the state belongs
+        state_name
+            The name of the state to validate
+        state_config
+            A ConfigTree defining the state to validate
+
+        Returns
+        -------
+        List[str]
+            A list of error messages
+        """
+        error_messages = []
+
+        if not isinstance(state_config, dict):
+            error_messages.append(
+                f"State configuration for in cause '{cause_name}' and "
+                f"state '{state_name}' must be a dictionary."
+            )
+            return error_messages
+
+        allowable_keys = set(CausesConfigurationParser.ALLOWABLE_STATE_KEYS)
+        if state_name in ["susceptible", "recovered"]:
+            allowable_keys.remove("data_sources")
+
+        if not set(state_config.keys()).issubset(allowable_keys):
+            error_messages.append(
+                f"State configuration for in cause '{cause_name}' and "
+                f"state '{state_name}' may only contain the following "
+                f"keys: {allowable_keys}."
+            )
+
+        state_type = state_config.get("state_type", "")
+        if state_type and not (isinstance(state_type, str) and "." in state_type):
+            error_messages.append(
+                f"If 'model_type' is provided for cause '{cause_name}' it "
+                "must be a fully qualified import path for the type of the "
+                f"desired model. Provided'{state_type}'."
+            )
+
+        if not isinstance(state_config.get("cause_type", ""), str):
+            error_messages.append(
+                f"Cause type for state '{state_name}' in cause '{cause_name}' "
+                f"must be a string. Provided {state_config['cause_type']}."
+            )
+        is_transient = state_config.get("transient", False)
```

Comment:
This is using the default value. It should refer to some canonical definition of defaults.

Reply:
This doesn't actually matter, since the value itself doesn't matter, merely the type